README.md

```diff
@@ -9,21 +9,621 @@ JavaScript Inheritance Patterns
+// {width: 2}
+console.log(rectangle);
+/*  { sides: 4, width: 5, height: 7,
+      area: [Function: area],
+      perimeter: [Function: perimeter] }
+*/
+```
+
+We already know that you can add properties to an object at runtime by
+just setting them, so here we're adding the property `width` to our
+`rect` oject, which means when we try to access it, JavaScript will find
+it on `rect` and will no longer have to access `__proto__`, effectively
+overriding the property for our `rect` object.
+
+A rectangle is really just a special quadrilateral, right? Let's build out this example a little more and see some inheritance in action.
+
+```js
+var quadrilateral = {
+  sides: 4,
+  sideOneLength: 1,
+  sideTwoLength: 2,
+  sideThreeLength: 1,
+  sideFourLength: 2,
+  perimeter: function() {
+    return this.sideOneLength + this.sideTwoLength + this.sideThreeLength + this.sideFourLength;
+  },
+  setSides: function(s1, s2, s3, s4) {
+    this.sideOneLength = s1;
+    this.sideTwoLength = s2;
+    this.sideThreeLength = s3;
+    this.sideFourLength = s4;
+  }
+}
+
+var rectangle = Object.create(quadrilateral);
+rectangle.width = function() {
+  return this.sideOneLength;
+}
+rectangle.height = function() {
+  return this.sideTwoLength;
+}
+rectangle.setWidth = function(width) {
+  this.sideOneLength = width;
+  this.sideThreeLength = width;
+}
+rectangle.setHeight = function(height) {
+  this.sideTwoLength = height;
+  this.sideFourLength = height;
+}
+rectangle.area = function() {
+  return this.width() * this.height();
+}
+
+var square = Object.create(rectangle);
+square.setSideLength = function(l) {
+  this.setWidth(l);
+  this.setHeight(l);
+}
+
+square.setSideLength(4);
+square.area();
+square.perimeter();
+```
+
+Now we have a base `quadrilateral` that we can use to build a
+`rectangle`. We'll extend the rectangle to have functions to determine
+more rectangular-specific things, like `width`, `height`, and `area`. It
+wouldn't make sense to define `area()` at the `quadrilateral` level,
+because it's a different formula depending on the shape. So we do that
+on `rectangle`. But we preserve `perimeter()` by using the side length
+properties of `quadrilateral`, so that even when we get down to
+`square`, we can still use the prototype system to delegate our
+perimeter formula up to `quadrilateral`.
+
+Since a square is just an equilateral rectangle, our `square` can use
+everything from the `rectangle` prototype, and just extend it to
+specifically set one side-length value instead of a `width` and
+`height`.
+
+This is the basic pattern of an inheritance chain. A parent object is
+more general than a child, which takes attributes from the parent and
+adds more specific attributes of its own.
 
 ## hasOwnProperty()
 
-- Make students "feel the pain" -- iterate through an object's properties and show that stuff is in there that shouldn't be (if it has inherited from another object)
+One interesting challenge with objects and inheritance is that we
```

Comment:
I was perhaps over-zealous here -- would it be easier to have a separate readme for hasOwnProperty()?